src/workspace-data/data-table/wds/WDSContent.test.ts

```diff
@@ -0,0 +1,229 @@
+  features: DataTableFeatures;
+  entities: Entity[];
+}
+
+const marbleSchema: RecordTypeSchema = {
+  name: 'marble',
+  count: 1,
+  attributes: [
+    { name: 'id', datatype: 'number' },
+    { name: 'color', datatype: 'string' },
+  ],
+  primaryKey: 'id',
+};
+const defaultProps: WDSContentProps = {
+  workspace: {
+    ...defaultAzureWorkspace,
+    workspace: {
+      ...defaultAzureWorkspace.workspace,
+      // attributes are required to avoid an error while destructuring from 'workspace-column-defaults'
+      attributes: {},
+    },
+  },
+  recordType: marbleSchema.name,
+  wdsSchema: [marbleSchema],
+  editable: true,
+  dataProvider: {} as DataTableProvider,
+};
+const defaultFeatures: DataTableFeatures = {
+  supportsCapabilities: true,
+  supportsTsvDownload: false,
+  supportsTsvAjaxDownload: false,
+  supportsTypeDeletion: false,
+  supportsTypeRenaming: false,
+  supportsEntityRenaming: false,
+  supportsEntityUpdating: false,
+  supportsAttributeRenaming: false,
+  supportsAttributeDeleting: false,
+  supportsAttributeClearing: false,
+  supportsExport: false,
+  supportsPointCorrection: false,
+  supportsFiltering: false,
+  supportsRowSelection: false,
+};
+
+const defaultSetupOptions: SetupOptions = {
+  props: defaultProps,
+  features: defaultFeatures as DataTableFeatures,
+  entities: [
+    {
+      name: '1',
+      entityType: 'marble',
+      attributes: { color: 'red' },
+    },
+    {
+      name: '2',
+      entityType: 'marble',
+      attributes: { color: 'yellow' },
+    },
+    {
+      name: '3',
+      entityType: 'marble',
+      attributes: { color: 'green' },
+    },
+  ],
+};
+describe('WDSContent', () => {
+  const setup = ({ props, features, entities }: SetupOptions = defaultSetupOptions) => {
+    const user = userEvent.setup();
+
+    const getPageResponse: DeepPartial<EntityQueryResponse> = {
+      results: entities,
+      resultMetadata: {
+        unfilteredCount: entities.length,
+        filteredCount: entities.length,
+        filteredPageCount: entities.length / 100 + 1,
+      },
+    };
+
+    const dataProvider: DeepPartial<DataTableProvider> = {
+      getPage: jest.fn().mockResolvedValue(getPageResponse),
+      features,
+    };
+
+    return { user, props: { ...props, dataProvider: dataProvider as DataTableProvider } };
+  };
+
+  describe('delete column button', () => {
+    it('is displayed when editable and supportsAttributeDeleting are true', async () => {
+      // Arrange
+      const { user, props } = setup({
+        ...defaultSetupOptions,
+        props: { ...defaultProps, editable: true },
+        features: { ...defaultFeatures, supportsAttributeDeleting: true },
+      });
+
+      await act(() => {
+        render(h(WDSContent, props));
+      });
+
+      const columnMenus = screen.queryAllByRole('button', { name: 'Column menu' });
+      expect(columnMenus.length).toEqual(2);
+      const [_unusedIdColumnMenu, colorColumnMenu] = columnMenus;
```

Comment:
this array destructuring depends on https://github.com/DataBiosphere/terra-ui/pull/4619